packages/core/base/src/signer.ts

```diff
@@ -48,13 +88,27 @@ export abstract class BaseSignerWalletAdapter extends BaseWalletAdapter implemen
 
     abstract signTransaction(transaction: Transaction): Promise<Transaction>;
 
+    signVersionedTransaction(transaction: VersionedTransaction): Promise<VersionedTransaction> {
```

Comment:
nit: make this `async` even though it doesn't `await` for consistency and to hint to the overrider how it should be implemented.